code/datums/diseases/advance/symptoms/shedding.dm

```diff
@@ -36,15 +36,16 @@ BONUS
 				if(3, 4)
 					if(!(head_organ.h_style == "Bald") && !(head_organ.h_style == "Balding Hair"))
 						to_chat(H, "<span class='warning'>Your hair starts to fall out in clumps...</span>")
-						spawn(50)
-							head_organ.h_style = "Balding Hair"
-							H.update_hair()
+						addtimer(CALLBACK(src, .proc/change_hair, H, head_organ, null, "Balding Hair"), 5 SECONDS)
 				if(5)
 					if(!(head_organ.f_style == "Shaved") || !(head_organ.h_style == "Bald"))
 						to_chat(H, "<span class='warning'>Your hair starts to fall out in clumps...</span>")
-						spawn(50)
-							head_organ.f_style = "Shaved"
-							head_organ.h_style = "Bald"
-							H.update_hair()
-							H.update_fhair()
-	return
+						addtimer(CALLBACK(src, .proc/change_hair, H, head_organ, "Shaved", "Bald"), 5 SECONDS)
+
+/datum/symptom/shedding/proc/change_hair(mob/living/carbon/human/H, obj/item/organ/external/head/head_organ, f_style, h_style)
```

Comment:
make sure to check for the head organ, again---and `H`; they could have gotten deleted.

Reply:
Can do.